From 25262086e789676d0f51acd463cb4a7a61b06a3d Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Mon, 22 Apr 2013 18:32:58 +0300
Subject: [PORT FROM MAIN] Telephony: Fix Locale construction when language code is null

BZ: 34594

When the language code is null, instead of constructing
with empty language code and valid country code, Locale
is constructed with country code provided as a Language
Code.

Framework calls the Locale with country code as language
code, this patch fixes this issue.

Change-Id: Id9020283c5024360f2f009589be57889e42bf41c
Orig-Change-Id: I83382f3c814d9fb3c762807eba86b440ef9d43b9
Category: aosp improvement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../com/android/internal/telephony/MccTable.java   |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/java/com/android/internal/telephony/MccTable.java b/src/java/com/android/internal/telephony/MccTable.java
index c57a8c4..09bcbcf 100644
--- a/src/java/com/android/internal/telephony/MccTable.java
+++ b/src/java/com/android/internal/telephony/MccTable.java
@@ -101,7 +101,7 @@ public final class MccTable
         } else {
             Locale locale;
             if (entry.mLanguage == null) {
-                locale = new Locale(entry.mIso);
+                locale = new Locale("", entry.mIso);
             } else {
                 locale = new Locale(entry.mLanguage, entry.mIso);
             }
-- 
1.7.4.1

